pkg/controllers/provisioning/scheduling/topology.go

```diff
@@ -278,6 +278,28 @@ func (t *Topology) countDomains(ctx context.Context, tg *TopologyGroup) error {
 		pods = append(pods, podList.Items...)
 	}
 
+	// capture new domain values from existing nodes that may not have any pods selected by the topology group
```

Comment:
After some discussion with @jmdeal, we should consider passing state nodes directly into this function rather than using the t.cluster.ForEach and holding a read lock throughout the time